espnet2/train/preprocessor.py

```diff
@@ -220,12 +225,27 @@ def __init__(
         else:
             self.noises = None
 
-    def _convolve_rir(self, speech, power, rirs):
+        # Check DataAugmentation docstring for more information of `data_aug_effects`
+        self.fs = fs
+        if data_aug_effects is not None:
+            assert self.fs > 0, self.fs
+            self.data_aug = DataAugmentation(data_aug_effects, apply_n=data_aug_num)
+        else:
+            self.data_aug = None
+        self.data_aug_prob = data_aug_prob
+
+    def _convolve_rir(self, speech, power, rirs, tgt_fs=None, single_channel=False):
         rir_path = np.random.choice(rirs)
         rir = None
         if rir_path is not None:
-            rir, _ = soundfile.read(rir_path, dtype=np.float64, always_2d=True)
+            rir, fs = soundfile.read(rir_path, dtype=np.float64, always_2d=True)
+            if tgt_fs and fs != tgt_fs:
```

Comment:
maybe better to warn of raise something because sample rate mismatch may not be intended

Reply:
Done